kv/kvserver: TestProtectedTimestamps failed

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11437663?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11437663?buildTab=artifacts#/) on master @ [c040b7338a946157b57ffe4f1207bff2bbd46576](https://github.com/cockroachdb/cockroach/commits/c040b7338a946157b57ffe4f1207bff2bbd46576):


```
        	            	     5.757ms      0.009ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (172 bytes)
        	            	     5.770ms      0.013ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     5.827ms      0.057ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     5.888ms      0.061ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     5.916ms      0.028ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     5.994ms      0.078ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     7.842ms      2.015ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     7.895ms      0.053ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1692717349.541737651,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1692717348.541737651,0 AffectedVersionsKeyBytes:47396 AffectedVersionsValBytes:91382276 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     7.931ms      0.036ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1692717351413340377 IntentAge:0 GCBytesAge:65053376 LiveBytes:32794 LiveCount:1 KeyBytes:15677 KeyCount:1 ValBytes:42806762 ValCount:1306 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:285 SysCount:6 AbortSpanBytes:0}
        	            	     7.952ms      0.021ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 1.46/fuzz(0.96)=1.52=valScaleScore(1.52)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 41 MiB non-live, curr. age 62 MiB*s, min exp. reduction: 21 MiB*s
        	            	     7.963ms      0.011ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     7.988ms      0.024ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 2.48/fuzz(0.96)=2.58=valScaleScore(2.58)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 330 MiB*s, min exp. reduction: 202 MiB*s
        	            	     8.005ms      0.017ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 1.46/fuzz(0.96)=1.52=valScaleScore(1.52)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 41 MiB non-live, curr. age 62 MiB*s, min exp. reduction: 21 MiB*s
        	            	     8.013ms      0.009ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     8.013ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 1ms}]
        	            	     8.013ms      0.000ms        [local proposal: {count: 1, duration 97µs}]
        	            	     8.037ms      0.023ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     8.067ms      0.031ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     8.085ms      0.018ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     8.117ms      0.032ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.126ms      0.010ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     8.151ms      0.024ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.152ms      0.002ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     8.152ms      0.000ms            [local proposal: {count: 1, duration 97µs}]
        	            	     8.161ms      0.009ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     8.186ms      0.025ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     8.198ms      0.012ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     8.210ms      0.013ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     8.217ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     8.231ms      0.013ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     8.239ms      0.008ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     9.264ms      1.026ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1692717351440167660 > , err=<nil>
        	            	     9.279ms      0.014ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	     9.301ms      0.022ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     9.312ms      0.011ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	     9.322ms      0.010ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (128 bytes)
        	            	     9.334ms      0.012ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     9.390ms      0.056ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     9.450ms      0.060ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     9.475ms      0.025ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     9.534ms      0.059ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     9.619ms      1.615ms    event:kv/kvserver/store.go:3602 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps1654883036
--- FAIL: TestProtectedTimestamps (19.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11476619?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11476619?buildTab=artifacts#/) on master @ [743a82f55c95eb2e49dd8e66562bc51c54a65ff4](https://github.com/cockroachdb/cockroach/commits/743a82f55c95eb2e49dd8e66562bc51c54a65ff4):


```
        	            	     5.791ms      0.011ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (172 bytes)
        	            	     5.805ms      0.014ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     5.862ms      0.057ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     5.920ms      0.058ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     5.945ms      0.025ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     6.002ms      0.057ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     7.996ms      2.134ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     8.054ms      0.058ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1692897589.267121720,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1692897588.267121720,0 AffectedVersionsKeyBytes:45305 AffectedVersionsValBytes:87350705 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     8.090ms      0.036ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1692897591069852990 IntentAge:0 GCBytesAge:86431804 LiveBytes:32794 LiveCount:1 KeyBytes:17153 KeyCount:1 ValBytes:46838333 ValCount:1429 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:285 SysCount:6 AbortSpanBytes:0}
        	            	     8.112ms      0.022ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 1.77/fuzz(0.96)=1.84=valScaleScore(1.85)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 45 MiB non-live, curr. age 82 MiB*s, min exp. reduction: 38 MiB*s
        	            	     8.123ms      0.011ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     8.150ms      0.027ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 2.65/fuzz(0.96)=2.76=valScaleScore(2.76)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 353 MiB*s, min exp. reduction: 225 MiB*s
        	            	     8.167ms      0.017ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 1.77/fuzz(0.96)=1.84=valScaleScore(1.85)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 45 MiB non-live, curr. age 82 MiB*s, min exp. reduction: 38 MiB*s
        	            	     8.180ms      0.013ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     8.180ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 2ms}]
        	            	     8.180ms      0.000ms        [local proposal: {count: 1, duration 111µs}]
        	            	     8.200ms      0.020ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     8.235ms      0.036ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     8.256ms      0.021ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     8.286ms      0.030ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.297ms      0.011ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     8.326ms      0.029ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.327ms      0.001ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     8.327ms      0.000ms            [local proposal: {count: 1, duration 111µs}]
        	            	     8.336ms      0.009ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     8.376ms      0.040ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     8.391ms      0.015ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     8.405ms      0.013ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     8.412ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     8.428ms      0.016ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     8.436ms      0.008ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     9.534ms      1.098ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1692897591095781749 > , err=<nil>
        	            	     9.551ms      0.017ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	     9.576ms      0.025ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     9.586ms      0.010ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	     9.597ms      0.011ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (128 bytes)
        	            	     9.614ms      0.017ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     9.677ms      0.063ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     9.753ms      0.076ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     9.782ms      0.029ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     9.849ms      0.067ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     9.935ms      1.768ms    event:kv/kvserver/store.go:3602 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps187617042
--- FAIL: TestProtectedTimestamps (19.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11484489?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11484489?buildTab=artifacts#/) on master @ [48fc58b6fb52f2e2fec8a6f5e97ec056b897b13c](https://github.com/cockroachdb/cockroach/commits/48fc58b6fb52f2e2fec8a6f5e97ec056b897b13c):


```
        	            	     6.236ms      0.011ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (171 bytes)
        	            	     6.250ms      0.014ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     6.307ms      0.057ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     6.383ms      0.075ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     6.411ms      0.028ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     6.500ms      0.089ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     8.299ms      1.992ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     8.357ms      0.058ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1692926974.427776905,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1692926973.427776905,0 AffectedVersionsKeyBytes:48144 AffectedVersionsValBytes:92824464 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     8.398ms      0.041ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1692926976251086460 IntentAge:0 GCBytesAge:68889689 LiveBytes:32794 LiveCount:1 KeyBytes:15149 KeyCount:1 ValBytes:41364574 ValCount:1262 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:285 SysCount:6 AbortSpanBytes:0}
        	            	     8.421ms      0.022ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 1.60/fuzz(0.96)=1.66=valScaleScore(1.67)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 39 MiB non-live, curr. age 66 MiB*s, min exp. reduction: 26 MiB*s
        	            	     8.433ms      0.012ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     8.459ms      0.027ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 2.62/fuzz(0.96)=2.72=valScaleScore(2.72)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 348 MiB*s, min exp. reduction: 220 MiB*s
        	            	     8.476ms      0.017ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 1.60/fuzz(0.96)=1.66=valScaleScore(1.67)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 39 MiB non-live, curr. age 66 MiB*s, min exp. reduction: 26 MiB*s
        	            	     8.489ms      0.013ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     8.489ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 2ms}]
        	            	     8.489ms      0.000ms        [local proposal: {count: 1, duration 87µs}]
        	            	     8.511ms      0.023ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     8.563ms      0.052ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     8.581ms      0.018ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     8.607ms      0.025ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.617ms      0.011ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     8.642ms      0.025ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.644ms      0.002ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     8.644ms      0.000ms            [local proposal: {count: 1, duration 87µs}]
        	            	     8.653ms      0.010ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     8.680ms      0.026ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     8.693ms      0.013ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     8.706ms      0.013ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     8.713ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     8.729ms      0.015ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     8.738ms      0.009ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     9.847ms      1.109ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1692926976280568884 > , err=<nil>
        	            	     9.862ms      0.015ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	     9.884ms      0.022ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     9.894ms      0.010ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	     9.905ms      0.011ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (127 bytes)
        	            	     9.918ms      0.014ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     9.982ms      0.063ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	    10.047ms      0.065ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	    10.076ms      0.028ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	    10.119ms      0.043ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	    10.214ms      1.738ms    event:kv/kvserver/store.go:3602 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps808377420
--- FAIL: TestProtectedTimestamps (19.89s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11486098?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11486098?buildTab=artifacts#/) on master @ [37f97b66d9ecfe4d9379e39f8fa90acf5f8d1ec1](https://github.com/cockroachdb/cockroach/commits/37f97b66d9ecfe4d9379e39f8fa90acf5f8d1ec1):


```
        	            	     5.943ms      0.010ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (172 bytes)
        	            	     5.956ms      0.014ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     6.015ms      0.059ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     6.110ms      0.095ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     6.152ms      0.043ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     6.212ms      0.059ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     7.853ms      1.838ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     7.928ms      0.075ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1692946999.649687417,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1692946998.649687417,0 AffectedVersionsKeyBytes:50490 AffectedVersionsValBytes:97368480 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     7.971ms      0.042ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1692947001031752152 IntentAge:0 GCBytesAge:63197892 LiveBytes:32801 LiveCount:1 KeyBytes:13481 KeyCount:1 ValBytes:36816432 ValCount:1123 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:270 SysCount:6 AbortSpanBytes:0}
        	            	     7.993ms      0.022ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 1.65/fuzz(0.96)=1.72=valScaleScore(1.72)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 35 MiB non-live, curr. age 60 MiB*s, min exp. reduction: 25 MiB*s
        	            	     8.003ms      0.010ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     8.026ms      0.023ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 2.68/fuzz(0.96)=2.78=valScaleScore(2.78)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 356 MiB*s, min exp. reduction: 228 MiB*s
        	            	     8.046ms      0.020ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 1.65/fuzz(0.96)=1.72=valScaleScore(1.72)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 35 MiB non-live, curr. age 60 MiB*s, min exp. reduction: 25 MiB*s
        	            	     8.057ms      0.010ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     8.057ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 1ms}]
        	            	     8.057ms      0.000ms        [local proposal: {count: 1, duration 81µs}]
        	            	     8.077ms      0.021ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     8.106ms      0.029ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     8.123ms      0.017ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     8.150ms      0.027ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.162ms      0.012ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     8.187ms      0.024ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.188ms      0.001ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     8.188ms      0.000ms            [local proposal: {count: 1, duration 81µs}]
        	            	     8.196ms      0.008ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     8.221ms      0.025ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     8.233ms      0.013ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     8.245ms      0.012ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     8.252ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     8.267ms      0.015ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     8.275ms      0.009ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     9.214ms      0.939ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1692947001058113736 > , err=<nil>
        	            	     9.229ms      0.015ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	     9.252ms      0.023ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     9.262ms      0.009ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	     9.273ms      0.012ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (128 bytes)
        	            	     9.286ms      0.013ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     9.346ms      0.060ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     9.406ms      0.060ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     9.433ms      0.027ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     9.473ms      0.040ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     9.559ms      1.513ms    event:kv/kvserver/store.go:3602 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps3366736850
--- FAIL: TestProtectedTimestamps (19.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11509830?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11509830?buildTab=artifacts#/) on master @ [671500be0c22d14d3cb0089313cb49c736631bac](https://github.com/cockroachdb/cockroach/commits/671500be0c22d14d3cb0089313cb49c736631bac):


```
        	            	     6.572ms      0.011ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (38613 bytes)
        	            	     6.589ms      0.018ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     6.656ms      0.067ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     6.743ms      0.087ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     6.773ms      0.030ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     7.168ms      0.395ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     7.822ms      1.166ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     7.895ms      0.073ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1693123288.175982960,2147483647 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1693123287.175982960,2147483647 AffectedVersionsKeyBytes:32691 AffectedVersionsValBytes:63030171 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:0 ClearRangeSpanFailures:0}
        	            	     7.932ms      0.037ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1693123291759843214 IntentAge:0 GCBytesAge:148796482 LiveBytes:32794 LiveCount:1 KeyBytes:26057 KeyCount:1 ValBytes:71158867 ValCount:2171 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:269 SysCount:6 AbortSpanBytes:0}
        	            	     7.953ms      0.021ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 2.01/fuzz(0.96)=2.09=valScaleScore(2.09)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 68 MiB non-live, curr. age 142 MiB*s, min exp. reduction: 74 MiB*s
        	            	     7.964ms      0.011ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     7.987ms      0.023ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 1.01/fuzz(0.96)=1.05=valScaleScore(1.05)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 134 MiB*s, min exp. reduction: 5.8 MiB*s
        	            	     8.005ms      0.018ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 2.01/fuzz(0.96)=2.09=valScaleScore(2.09)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 68 MiB non-live, curr. age 142 MiB*s, min exp. reduction: 74 MiB*s
        	            	     8.015ms      0.010ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     8.015ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 2ms}]
        	            	     8.015ms      0.000ms        [local proposal: {count: 1, duration 95µs}]
        	            	     8.036ms      0.021ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     8.068ms      0.033ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     8.086ms      0.018ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     8.117ms      0.031ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.128ms      0.011ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     8.154ms      0.026ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.155ms      0.001ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     8.155ms      0.000ms            [local proposal: {count: 1, duration 95µs}]
        	            	     8.164ms      0.008ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     8.189ms      0.025ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     8.201ms      0.013ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     8.214ms      0.012ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     8.221ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     8.235ms      0.015ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     8.243ms      0.008ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	    10.306ms      2.063ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1693123291789214293 > , err=<nil>
        	            	    10.322ms      0.016ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	    10.344ms      0.023ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	    10.354ms      0.010ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	    10.364ms      0.010ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (128 bytes)
        	            	    10.379ms      0.016ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	    10.436ms      0.057ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	    10.496ms      0.060ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	    10.522ms      0.026ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	    10.579ms      0.057ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	    10.654ms      2.649ms    event:kv/kvserver/store.go:3607 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps3711627293
--- FAIL: TestProtectedTimestamps (19.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11547883?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11547883?buildTab=artifacts#/) on master @ [36bd8296bad475a8183e474bf4aeb08bb83f8a2d](https://github.com/cockroachdb/cockroach/commits/36bd8296bad475a8183e474bf4aeb08bb83f8a2d):


```
        	            	     1.426ms      0.029ms    event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     1.438ms      0.012ms    event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     1.446ms      0.008ms    event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     1.462ms      0.016ms    event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     1.471ms      0.009ms    event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     1.490ms      0.019ms    event:storage/mvcc.go:5320 [n1,s1,r65/1:/{Table/104-Max}] done with GC evaluation for 0 keys at 0.00 keys/sec. Deleted 0 entries
        	            	     1.497ms      0.007ms    event:storage/mvcc.go:5320 [n1,s1,r65/1:/{Table/104-Max}] done with GC evaluation for 0 keys at 0.00 keys/sec. Deleted 0 entries
        	            	     1.506ms      0.009ms    event:storage/mvcc.go:5628 [n1,s1,r65/1:/{Table/104-Max}] done with GC evaluation for 0 range keys at 0.00 keys/sec. Deleted 0 entries
        	            	     1.625ms      0.119ms    event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated GC command header:<key:"\360" end_key:"\377\377" > threshold:<wall_time:1693355493232672190 > , txn=<nil> : resp=header:<> , err=<nil>
        	            	     1.639ms      0.014ms    event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=1
        	            	     1.660ms      0.021ms    event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     1.670ms      0.009ms    event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=55 bytes
        	            	     1.679ms      0.009ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (175 bytes)
        	            	     1.693ms      0.014ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     1.746ms      0.054ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     1.818ms      0.072ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     1.847ms      0.028ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     1.907ms      0.061ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     3.880ms      2.134ms    event:kv/kvserver/mvcc_gc_queue.go:562 [n1,s1,r65/1:/{Table/104-Max}] sending batch 2 (0 keys)
        	            	     3.921ms      0.040ms    event:kv/kvserver/kvflowcontrol/kvflowcontroller/kvflowcontroller.go:158 [n1,s1,r65/1:/{Table/104-Max}] admitted request (pri=bulk-normal-pri stream=t1/s1 tokens=+8.0 MiB wait-duration=2.55µs mode=apply_to_all)
        	            	     3.950ms      0.030ms    event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     3.963ms      0.012ms    event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     3.970ms      0.007ms    event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     3.986ms      0.016ms    event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     3.994ms      0.008ms    event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     4.013ms      0.019ms    event:storage/mvcc.go:5320 [n1,s1,r65/1:/{Table/104-Max}] done with GC evaluation for 0 keys at 0.00 keys/sec. Deleted 0 entries
        	            	     4.023ms      0.010ms    event:storage/mvcc.go:5320 [n1,s1,r65/1:/{Table/104-Max}] done with GC evaluation for 0 keys at 0.00 keys/sec. Deleted 0 entries
        	            	     5.893ms      1.870ms    event:storage/mvcc.go:5903 [n1,s1,r65/1:/{Table/104-Max}] done with GC evaluation for clear range of 4093 keys at 2220192.30 keys/sec. Deleted 2833 entries
        	            	     5.907ms      0.014ms    event:storage/mvcc.go:5628 [n1,s1,r65/1:/{Table/104-Max}] done with GC evaluation for 0 range keys at 0.00 keys/sec. Deleted 0 entries
        	            	     6.009ms      0.102ms    event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated GC command header:<key:"\360" end_key:"\377\377" > threshold:<> clear_range:<start_key:"\360\211\211\210" end_key:"\360\211\211\210\000" start_key_timestamp:<wall_time:1693355493230639063 > > , txn=<nil> : resp=header:<> , err=<nil>
        	            	     6.024ms      0.015ms    event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=1
        	            	     6.048ms      0.024ms    event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     6.057ms      0.010ms    event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, -2833 new values, 0 new intents, write batch size=37 bytes
        	            	     6.067ms      0.010ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (171 bytes)
        	            	     6.085ms      0.018ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     6.143ms      0.059ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     6.211ms      0.067ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     6.237ms      0.026ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     6.303ms      0.066ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     8.118ms      1.975ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     8.172ms      0.054ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1693355494.232672190,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1693355493.232672190,0 AffectedVersionsKeyBytes:48161 AffectedVersionsValBytes:92877072 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     8.206ms      0.034ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1693355495950803851 IntentAge:0 GCBytesAge:38272932 LiveBytes:32801 LiveCount:1 KeyBytes:15125 KeyCount:1 ValBytes:41307840 ValCount:1260 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:292 SysCount:6 AbortSpanBytes:0}
        	            	     8.226ms      0.019ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=false with 0.89/fuzz(0.96)=0.93=valScaleScore(0.93)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 39 MiB non-live, curr. age 36 MiB*s, min exp. reduction: 0 B*s
        	            	     8.239ms      0.014ms    event:kv/kvserver/store.go:3607 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps865362694
--- FAIL: TestProtectedTimestamps (19.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11571127?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11571127?buildTab=artifacts#/) on master @ [250f2f77fb805e752a1033cce4313d7c7616c162](https://github.com/cockroachdb/cockroach/commits/250f2f77fb805e752a1033cce4313d7c7616c162):


```
        	            	     5.716ms      0.010ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (171 bytes)
        	            	     5.729ms      0.014ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     5.785ms      0.056ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     5.847ms      0.062ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     5.877ms      0.030ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     5.947ms      0.069ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     8.279ms      2.494ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     8.367ms      0.088ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1693494110.741413731,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1693494109.741413731,0 AffectedVersionsKeyBytes:43367 AffectedVersionsValBytes:83614127 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     8.407ms      0.039ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1693494112740534803 IntentAge:0 GCBytesAge:63086036 LiveBytes:32794 LiveCount:1 KeyBytes:18521 KeyCount:1 ValBytes:50574911 ValCount:1543 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:285 SysCount:6 AbortSpanBytes:0}
        	            	     8.429ms      0.022ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 1.20/fuzz(0.96)=1.25=valScaleScore(1.25)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 48 MiB non-live, curr. age 60 MiB*s, min exp. reduction: 12 MiB*s
        	            	     8.440ms      0.010ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     8.463ms      0.023ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 2.05/fuzz(0.96)=2.13=valScaleScore(2.13)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 273 MiB*s, min exp. reduction: 145 MiB*s
        	            	     8.483ms      0.020ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 1.20/fuzz(0.96)=1.25=valScaleScore(1.25)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 48 MiB non-live, curr. age 60 MiB*s, min exp. reduction: 12 MiB*s
        	            	     8.494ms      0.011ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     8.494ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 2ms}]
        	            	     8.494ms      0.000ms        [local proposal: {count: 1, duration 79µs}]
        	            	     8.516ms      0.021ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     8.546ms      0.031ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     8.564ms      0.018ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     8.593ms      0.029ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.603ms      0.009ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     8.627ms      0.024ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     8.628ms      0.001ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     8.628ms      0.000ms            [local proposal: {count: 1, duration 79µs}]
        	            	     8.636ms      0.008ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     8.664ms      0.028ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     8.677ms      0.014ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     8.689ms      0.012ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     8.696ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     8.710ms      0.014ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     8.718ms      0.008ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     9.906ms      1.188ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1693494112768806750 > , err=<nil>
        	            	     9.919ms      0.013ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	     9.942ms      0.023ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     9.952ms      0.010ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	     9.965ms      0.013ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (128 bytes)
        	            	     9.979ms      0.014ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	    10.036ms      0.057ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	    10.096ms      0.060ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	    10.121ms      0.026ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	    10.160ms      0.039ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	    10.240ms      1.757ms    event:kv/kvserver/store.go:3607 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps2223925194
--- FAIL: TestProtectedTimestamps (20.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11579151?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11579151?buildTab=artifacts#/) on master @ [5554b75ae1c8a724289efa99e06bb7ba7c368292](https://github.com/cockroachdb/cockroach/commits/5554b75ae1c8a724289efa99e06bb7ba7c368292):


```
        	            	     5.834ms      0.010ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (172 bytes)
        	            	     5.847ms      0.013ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     5.881ms      0.034ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     5.943ms      0.062ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     5.968ms      0.025ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     6.025ms      0.057ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     7.579ms      1.698ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     7.656ms      0.077ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1693518440.269066380,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1693518439.269066380,0 AffectedVersionsKeyBytes:50830 AffectedVersionsValBytes:98003230 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     7.694ms      0.038ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1693518441479305931 IntentAge:0 GCBytesAge:40199314 LiveBytes:32794 LiveCount:1 KeyBytes:13253 KeyCount:1 ValBytes:36185808 ValCount:1104 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:263 SysCount:6 AbortSpanBytes:0}
        	            	     7.726ms      0.032ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 1.07/fuzz(0.96)=1.11=valScaleScore(1.11)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 34 MiB non-live, curr. age 38 MiB*s, min exp. reduction: 3.8 MiB*s
        	            	     7.738ms      0.011ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     7.762ms      0.024ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 3.06/fuzz(0.96)=3.19=valScaleScore(3.19)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 408 MiB*s, min exp. reduction: 280 MiB*s
        	            	     7.778ms      0.016ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 1.07/fuzz(0.96)=1.11=valScaleScore(1.11)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 34 MiB non-live, curr. age 38 MiB*s, min exp. reduction: 3.8 MiB*s
        	            	     7.786ms      0.009ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     7.786ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 1ms}]
        	            	     7.786ms      0.000ms        [local proposal: {count: 1, duration 86µs}]
        	            	     7.806ms      0.020ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     7.835ms      0.029ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     7.851ms      0.016ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     7.877ms      0.025ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     7.886ms      0.009ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     7.910ms      0.024ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     7.911ms      0.001ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     7.911ms      0.000ms            [local proposal: {count: 1, duration 86µs}]
        	            	     7.919ms      0.008ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     7.942ms      0.023ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     7.954ms      0.011ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     7.965ms      0.011ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     7.971ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     7.985ms      0.013ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     7.992ms      0.008ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     8.972ms      0.979ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1693518441505271566 > , err=<nil>
        	            	     8.986ms      0.014ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	     9.007ms      0.022ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     9.017ms      0.009ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	     9.027ms      0.010ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (128 bytes)
        	            	     9.040ms      0.013ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     9.080ms      0.040ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     9.142ms      0.062ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     9.169ms      0.027ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     9.215ms      0.046ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     9.296ms      1.518ms    event:kv/kvserver/store.go:3607 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps4161165260
--- FAIL: TestProtectedTimestamps (19.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestProtectedTimestamps [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11641027?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11641027?buildTab=artifacts#/) on master @ [1f95b88c42d136fe670a79458233a9ae768d2cde](https://github.com/cockroachdb/cockroach/commits/1f95b88c42d136fe670a79458233a9ae768d2cde):


```
        	            	     5.763ms      0.012ms    event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (172 bytes)
        	            	     5.776ms      0.014ms    event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     5.819ms      0.043ms    event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     5.887ms      0.068ms        === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     5.920ms      0.032ms        event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     5.978ms      0.058ms        event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.GCResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     7.596ms      1.777ms    event:kv/kvserver/gc/gc.go:379 [n1,s1,r65/1:/{Table/104-Max}] processing AbortSpan
        	            	     7.652ms      0.056ms    event:kv/kvserver/gc/gc.go:387 [n1,s1,r65/1:/{Table/104-Max}] GC'ed keys; stats {Now:1693943171.473577107,0 GCTTL:1s NumKeysAffected:1 NumRangeKeysAffected:0 IntentsConsidered:0 IntentTxns:0 TransactionSpanTotal:0 TransactionSpanGCAborted:0 TransactionSpanGCCommitted:0 TransactionSpanGCStaging:0 TransactionSpanGCPending:0 AbortSpanTotal:0 AbortSpanConsidered:0 AbortSpanGCNum:0 PushTxn:0 ResolveTotal:0 Threshold:1693943170.473577107,0 AffectedVersionsKeyBytes:48297 AffectedVersionsValBytes:93139344 AffectedVersionsRangeKeyBytes:0 AffectedVersionsRangeValBytes:0 ClearRangeSpanOperations:1 ClearRangeSpanFailures:0}
        	            	     7.711ms      0.059ms    event:kv/kvserver/mvcc_gc_queue.go:777 [n1,s1,r65/1:/{Table/104-Max}] MVCC stats after GC: {ContainsEstimates:0 LastUpdateNanos:1693943173253023265 IntentAge:0 GCBytesAge:65657592 LiveBytes:32801 LiveCount:1 KeyBytes:15029 KeyCount:1 ValBytes:41045568 ValCount:1252 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:292 SysCount:6 AbortSpanBytes:0}
        	            	     7.734ms      0.023ms    event:kv/kvserver/mvcc_gc_queue.go:778 [n1,s1,r65/1:/{Table/104-Max}] GC score after GC: queue=true with 1.54/fuzz(0.96)=1.60=valScaleScore(1.60)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 39 MiB non-live, curr. age 63 MiB*s, min exp. reduction: 24 MiB*s
        	            	     7.745ms      0.011ms    event:kv/kvserver/mvcc_gc_queue.go:794 [n1,s1,r65/1:/{Table/104-Max}] GC still needed following GC, recomputing MVCC stats
        	            	     7.769ms      0.023ms    event:kv/kvserver/mvcc_gc_queue.go:795 [n1,s1,r65/1:/{Table/104-Max}] old score queue=true with 2.68/fuzz(0.96)=2.79=valScaleScore(2.79)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 128 MiB non-live, curr. age 357 MiB*s, min exp. reduction: 229 MiB*s
        	            	     7.787ms      0.019ms    event:kv/kvserver/mvcc_gc_queue.go:796 [n1,s1,r65/1:/{Table/104-Max}] new score queue=true with 1.54/fuzz(0.96)=1.60=valScaleScore(1.60)*deadFrac(1.00)+intentScore(0.00)
        	            	likely last GC: never, 39 MiB non-live, curr. age 63 MiB*s, min exp. reduction: 24 MiB*s
        	            	     7.796ms      0.009ms        === operation:dist sender send _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max}
        	            	     7.796ms      0.000ms        [/cockroach.roachpb.Internal/Batch: {count: 1, duration 1ms}]
        	            	     7.796ms      0.000ms        [local proposal: {count: 1, duration 70µs}]
        	            	     7.819ms      0.023ms        event:kv/kvclient/kvcoord/range_iter.go:188 [n1,s1,r65/1:/{Table/104-Max}] querying next range at /Table/104
        	            	     7.866ms      0.046ms        event:kv/kvclient/kvcoord/range_iter.go:225 [n1,s1,r65/1:/{Table/104-Max}] key: /Table/104, desc: r65:/{Table/104-Max} [(n1,s1):1, next=2, gen=1]
        	            	     7.884ms      0.019ms        event:kv/kvclient/kvcoord/dist_sender.go:2248 [n1,s1,r65/1:/{Table/104-Max}] r65: sending batch 1 RecomputeStats to (n1,s1):1
        	            	     7.910ms      0.026ms        event:kv/kvclient/kvcoord/dist_sender.go:2563 [n1,s1,r65/1:/{Table/104-Max}] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     7.920ms      0.010ms        event:rpc/nodedialer/nodedialer.go:138 [n1,s1,r65/1:/{Table/104-Max}] sending request to local client
        	            	     7.947ms      0.027ms        event:server/node.go:1412 [n1] unable to determine if the given nodes are cross zone localities must have a valid zone tier key for cross-zone comparison
        	            	     7.949ms      0.002ms            === operation:/cockroach.roachpb.Internal/Batch _verbose:1 node:1 span.kind:server request:RecomputeStats [/Table/104,/Min)
        	            	     7.949ms      0.000ms            [local proposal: {count: 1, duration 70µs}]
        	            	     7.957ms      0.008ms            event:server/node.go:1320 [n1] node received request: 1 RecomputeStats
        	            	     7.985ms      0.028ms            event:kv/kvserver/store_send.go:154 [n1,s1] executing RecomputeStats [/Table/104,/Min)
        	            	     7.998ms      0.013ms            event:kv/kvserver/replica_send.go:187 [n1,s1,r65/1:/{Table/104-Max}] read-write path
        	            	     8.010ms      0.012ms            event:kv/kvserver/concurrency/concurrency_manager.go:209 [n1,s1,r65/1:/{Table/104-Max}] sequencing request
        	            	     8.017ms      0.007ms            event:kv/kvserver/concurrency/concurrency_manager.go:290 [n1,s1,r65/1:/{Table/104-Max}] acquiring latches
        	            	     8.034ms      0.017ms            event:kv/kvserver/replica_write.go:167 [n1,s1,r65/1:/{Table/104-Max}] applied timestamp cache
        	            	     8.042ms      0.008ms            event:kv/kvserver/replica_write.go:407 [n1,s1,r65/1:/{Table/104-Max}] executing read-write batch
        	            	     9.070ms      1.027ms            event:kv/kvserver/replica_evaluate.go:508 [n1,s1,r65/1:/{Table/104-Max}] evaluated RecomputeStats command header:<key:"\360" > , txn=<nil> : resp=header:<> added_delta:<last_update_nanos:1693943173280271073 > , err=<nil>
        	            	     9.084ms      0.015ms            event:kv/kvserver/replica_proposal.go:965 [n1,s1,r65/1:/{Table/104-Max}] need consensus on write batch with op count=0
        	            	     9.108ms      0.024ms            event:kv/kvserver/replica_raft.go:125 [n1,s1,r65/1:/{Table/104-Max}] evaluated request
        	            	     9.118ms      0.010ms            event:kv/kvserver/replica_raft.go:170 [n1,s1,r65/1:/{Table/104-Max}] proposing command to write 0 new keys, 0 new values, 0 new intents, write batch size=14 bytes
        	            	     9.128ms      0.010ms            event:kv/kvserver/replica_raft.go:275 [n1,s1,r65/1:/{Table/104-Max}] acquiring proposal quota (128 bytes)
        	            	     9.142ms      0.013ms            event:kv/kvserver/replica_raft.go:431 [n1,s1,r65/1:/{Table/104-Max}] submitting proposal to proposal buffer
        	            	     9.180ms      0.038ms            event:kv/kvserver/replica_proposal_buf.go:601 [n1,s1,r65/1:/{Table/104-Max}] flushing proposal to Raft
        	            	     9.234ms      0.054ms                === operation:local proposal _verbose:1 node:1 store:1 range:65/1:/{Table/104-Max} raft:
        	            	     9.258ms      0.024ms                event:kv/kvserver/app_batch.go:121 [n1,s1,r65/1:/{Table/104-Max},raft] applying command
        	            	     9.298ms      0.041ms                event:kv/kvserver/replica_application_state_machine.go:176 [n1,s1,r65/1:/{Table/104-Max},raft] LocalResult (reply: (err: <nil>), *kvpb.RecomputeStatsResponse, #encountered intents: 0, #acquired locks: 0, #resolved locks: 0#updated txns: 0 #end txns: 0, GossipFirstRange:false MaybeGossipSystemConfig:false MaybeGossipSystemConfigIfHaveFailure:false MaybeAddToSplitQueue:false MaybeGossipNodeLiveness:<nil> 
        	            	     9.362ms      1.575ms    event:kv/kvserver/store.go:3607 [n1,s1,r65/1:/{Table/104-Max}] processed: true (err: <nil>)
        	            	" to match "(?s)shouldQueue=false"
        	Test:       	TestProtectedTimestamps
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d48402a71deafe4adff22980d19f276/logTestProtectedTimestamps726752994
--- FAIL: TestProtectedTimestamps (19.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestamps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

This test also fails with:

```
--- FAIL: TestProtectedTimestamps (17.82s)
    test_log_scope.go:167: test logs captured to: /Users/pavel/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/577bfc8ed46850d1aaafa923fa37ce78/logTestProtectedTimestamps302068808
    test_log_scope.go:81: use -show-logs to present logs inline
    client_protectedts_test.go:203: 
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_protectedts_test.go:203
                Error:          Should be true
                Test:           TestProtectedTimestamps
                Messages:       threshold: 1693945517.233710000,0, protected 1693945516.544688000,0 "..."
```

--- a/pkg/kv/kvserver/client_protectedts_test.go
+++ b/pkg/kv/kvserver/client_protectedts_test.go
@@ -183,11 +183,8 @@ ORDER BY raw_start_key ASC LIMIT 1`)
 	// Verify that the record did indeed make its way down into KV where the
 	// replica can read it from.
 	ptsReader := tc.GetFirstStoreFromServer(t, 0).GetStoreConfig().ProtectedTimestampReader
-	require.NoError(
-		t,
-		ptutil.TestingVerifyProtectionTimestampExistsOnSpans(
-			ctx, t, s0, ptsReader, ptsRec.Timestamp, ptsRec.DeprecatedSpans,
-		),
+	ptutil.TestingWaitForProtectedTimestampToExistOnSpans(
+		ctx, t, s0, ptsReader, ptsRec.Timestamp, ptsRec.DeprecatedSpans,
 	)
 	upsertUntilBackpressure()
 	// We need to be careful choosing a time. We're a little limited because the
@@ -223,11 +220,8 @@ ORDER BY raw_start_key ASC LIMIT 1`)
 	// Verify that the record did indeed make its way down into KV where the
 	// replica can read it from. We then verify (below) that the failed record
 	// does not affect the ability to GC.
-	require.NoError(
-		t,
-		ptutil.TestingVerifyProtectionTimestampExistsOnSpans(
-			ctx, t, s0, ptsReader, failedRec.Timestamp, failedRec.DeprecatedSpans,
-		),
+	ptutil.TestingWaitForProtectedTimestampToExistOnSpans(
+		ctx, t, s0, ptsReader, failedRec.Timestamp, failedRec.DeprecatedSpans,
 	)
 
 	// Add a new record that is after the old record.
@@ -236,11 +230,8 @@ ORDER BY raw_start_key ASC LIMIT 1`)
 	laterRec.Timestamp = afterWrites
 	laterRec.Timestamp.Logical = 0
 	require.NoError(t, ptsWithDB.Protect(ctx, &laterRec))
-	require.NoError(
-		t,
-		ptutil.TestingVerifyProtectionTimestampExistsOnSpans(
-			ctx, t, s0, ptsReader, laterRec.Timestamp, laterRec.DeprecatedSpans,
-		),
+	ptutil.TestingWaitForProtectedTimestampToExistOnSpans(
+		ctx, t, s0, ptsReader, laterRec.Timestamp, laterRec.DeprecatedSpans,
 	)
 
 	// Release the record that had succeeded and ensure that GC eventually

--- a/pkg/kv/kvserver/client_protectedts_test.go
+++ b/pkg/kv/kvserver/client_protectedts_test.go
@@ -180,6 +180,15 @@ ORDER BY raw_start_key ASC LIMIT 1`)
 		Target:    ptpb.MakeSchemaObjectsTarget([]descpb.ID{getTableID()}),
 	}
 	require.NoError(t, ptsWithDB.Protect(ctx, &ptsRec))
+	// Verify that the record did indeed make its way down into KV where the
+	// replica can read it from.
+	ptsReader := tc.GetFirstStoreFromServer(t, 0).GetStoreConfig().ProtectedTimestampReader
+	require.NoError(
+		t,
+		ptutil.TestingVerifyProtectionTimestampExistsOnSpans(
+			ctx, t, s0, ptsReader, ptsRec.Timestamp, ptsRec.DeprecatedSpans,
+		),
+	)
 	upsertUntilBackpressure()
 	// We need to be careful choosing a time. We're a little limited because the
 	// ttl is defined in seconds and we need to wait for the threshold to be
@@ -201,16 +210,6 @@ ORDER BY raw_start_key ASC LIMIT 1`)
 	require.Regexp(t, "(?s)done with GC evaluation for 0 keys", trace.String())
 	thresh := thresholdFromTrace(trace)
 	require.Truef(t, thresh.Less(ptsRec.Timestamp), "threshold: %v, protected %v %q", thresh, ptsRec.Timestamp, trace)
-
-	// Verify that the record did indeed make its way down into KV where the
-	// replica can read it from.
-	ptsReader := tc.GetFirstStoreFromServer(t, 0).GetStoreConfig().ProtectedTimestampReader
-	require.NoError(
-		t,
-		ptutil.TestingVerifyProtectionTimestampExistsOnSpans(
-			ctx, t, s0, ptsReader, ptsRec.Timestamp, ptsRec.DeprecatedSpans,
-		),
-	)
 
 	// Make a new record that is doomed to fail.
 	failedRec := ptsRec

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -4228,15 +4228,9 @@ func TestStrictGCEnforcement(t *testing.T) {
 				}
 				ptsReader := tc.GetFirstStoreFromServer(t, i).GetStoreConfig().ProtectedTimestampReader
 				_, r := getFirstStoreReplica(t, tc.Server(i), tableKey)
-				testutils.SucceedsSoon(t, func() error {
-					if err := ptutil.TestingVerifyProtectionTimestampExistsOnSpans(ctx, t, tc.Server(i),
-						ptsReader, protectionTimestamp,
-						[]roachpb.Span{span}); err != nil {
-						return errors.Newf("protection timestamp %s does not exist on span %s",
-							protectionTimestamp, span)
-					}
-					return nil
-				})
+				ptutil.TestingWaitForProtectedTimestampToExistOnSpans(ctx, t, tc.Server(i),
+					ptsReader, protectionTimestamp,
+					[]roachpb.Span{span})
 				require.NoError(t, r.ReadProtectedTimestampsForTesting(ctx))
 			}
 		}

--- a/pkg/kv/kvserver/protectedts/ptutil/testutils.go
+++ b/pkg/kv/kvserver/protectedts/ptutil/testutils.go
@@ -27,17 +27,17 @@ import (
 	"github.com/cockroachdb/errors"
 )
 
-// TestingVerifyProtectionTimestampExistsOnSpans refreshes the PTS state in KV and
-// ensures a protection at the given protectionTimestamp exists for all the
-// supplied spans.
-func TestingVerifyProtectionTimestampExistsOnSpans(
+// TestingWaitForProtectedTimestampToExistOnSpans waits
+// testutils.SucceedsSoonDuration for the supplied protected timestamp to exist
+// on all the supplied spans. It fatals if this doesn't happen in time.
+func TestingWaitForProtectedTimestampToExistOnSpans(
 	ctx context.Context,
 	t *testing.T,
 	srv serverutils.TestServerInterface,
 	ptsReader spanconfig.ProtectedTSReader,
-	protectionTimestamp hlc.Timestamp,
+	protectedTimestamp hlc.Timestamp,
 	spans roachpb.Spans,
-) error {
+) {
 	testutils.SucceedsSoon(t, func() error {
 		if err := spanconfigptsreader.TestingRefreshPTSState(
 			ctx, t, ptsReader, srv.Clock().Now(),
@@ -51,18 +51,17 @@ func TestingVerifyProtectionTimestampExistsOnSpans(
 			}
 			found := false
 			for _, ts := range timestamps {
-				if ts.Equal(protectionTimestamp) {
+				if ts.Equal(protectedTimestamp) {
 					found = true
 					break
 				}
 			}
 			if !found {
-				return errors.Newf("protection timestamp %s does not exist on span %s", protectionTimestamp, sp)
+				return errors.Newf("protection timestamp %s does not exist on span %s", protectedTimestamp, sp)
 			}
 		}
 		return nil
 	})
-	return nil
 }
 
 func GetPTSTarget(t *testing.T, db *sqlutils.SQLRunner, ptsID *uuid.UUID) *ptpb.Target {

